common/src/main/org/jetbrains/lincheck/util/AnalysisSections.kt

```diff
@@ -168,7 +168,7 @@ inline fun <R> runInsideIgnoredSection(block: () -> R): R {
  * @return result of the [block] invocation or `null` if the analysis is disabled for the current thread.
  */
 inline fun <R> runInsideInjectedCode(block: () -> R): R? {
-    val desc = ThreadDescriptor.getCurrentThreadDescriptor() ?: return block()
+    val desc = ThreadDescriptor.getCurrentThreadDescriptor() ?: return null
```

Comment:
Could you please re-check. I think this method is used incorrectly for `TraceCollectingEventTracker::onMethodCallReturn`.

For `onMethodCallException` it will use `t: Throwable` as default value, but for `onMethodCallReturn` it will return `null`.

Reply:
Methods `TraceCollectingEventTracker::onMethodCall[Return/Exception]` use another overload of this function:
```kotlin
inline fun <R> runInsideInjectedCode(default: R, block: () -> R): R {
    return runInsideInjectedCode(block) ?: default
}
```
and it would return `default` in such case (which is the `result: Any?` or `t: Throwable`)